request.el

```diff
@@ -948,7 +910,20 @@ as the `--user' option."
    (list request-curl
          "--silent" "--location"
          "--cookie" cookie-jar "--cookie-jar" cookie-jar)
-   (request--auth auth url)
+   (when auth
+     (let* ((host (url-host (url-generic-parse-url url)))
+            (auth-source-do-cache nil)
+            (auth-source-creation-prompts '((user . (format "%s user: " host))
+                                            (secret . "Password for %u: ")))
+            (cred (car (auth-source-search
+                        :host host :require '(:user :secret) :create t :max 1))))
+       (split-string (format "--%s --user %s:%s"
+                             auth
```

Comment:
I don't think `auth` is correct here. I think this is required:
```
                             (or (plist-get auth :type) "anyauth")
```

Reply:
the intention is to call something like:
```
(request
  "http://example.com"
  :auth "digest"
  :success (lambda (&rest _args)
             (message "success and nothing less")))
```
where  `:auth` is the type.